packages/front-end/components/Experiment/TabbedPage/HealthTab.tsx

```diff
@@ -52,9 +54,11 @@ export default function HealthTab({
   );
 
   const hasPermissionToConfigHealthTag =
-    permissions.check("organizationSettings") &&
-    permissions.check("runQueries", datasource?.projects || []) &&
-    permissions.check("editDatasourceSettings", datasource?.projects || []);
+    (datasource &&
+      permissions.check("organizationSettings") &&
+      permissions.check("runQueries", datasource.projects) &&
+      permissionsUtil.canUpdateDataSourceSettings(datasource)) ||
+    false;
```

Comment:
@msamper - I updated this logic so if we `datasource` is undefined, it calculates to `false`.

It looks like we'll never show the Health Tab if the `datasource` is undefined, so this feels like a safe fix and makes the code a bit easier to read.